Automattic/jetpack
# phan: Allow deleting baseline files

## Proposed changes:

In many cases it's pointless to keep around the empty baseline files, ideally they should never be repopulated. Update the tooling to allow for these to be deleted, in which case they won't be re-created by `--update-baseline`.

But in case for some reason one does need to be re-created, add a `--force-update-baseline` option to allow for that.

Also update the GitHub Action checking for baseline changes to handle these, and also to handle changes in pseudo-project baselines that wasn't done in #36467.

For the moment I'm only actually removing the empty baselines for stuff Garage maintains. We can look at the rest in a followup.

### Other information:


## Jetpack product discussion


None

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Is CI happy?
* Try running `jetpack phan --update-baseline` for one of the projects with a removed baseline. It shouldn't re-create it.
* Try running `jetpack phan --force-update-baseline` for one of the projects with a removed baseline. It still shouldn't re-create it.
* Edit one of the projects with a removed baseline to add a Phan issue (e.g. change a class name in a phpdoc comment somewhere to some nonexistent class).
  * Try running `jetpack phan --update-baseline` for the project. It shouldn't re-create it.
  * Try running `jetpack phan --force-update-baseline` for the project. It should re-create it.
* Create a PR based on this one that will need updates to some of the baseline files, both ones that are empty and ones that aren't empty. Verify that the action produces appropriate messages about needing to regenerate those baselines.
  * A simple way to make a baseline file need updates is to change the "This is an automatically generated baseline for Phan issues" doc comment at the top.
